library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> Re: has_to_pexpr. It does something different on pre-expressions than has_quote. The has_quote instance would be mk_quote.

Yes, I understood that class to be more of a conversion, so I made up a new one.

`itactic` feels like a special and quite different beast to me than the other grammar items. It is not reflected like all the other ones and it needs additional info in form of the current `tac_class`. I assume for these reasons it is not in the `interactive.types` namespace right now (which is why I missed it) and likewise I'm not sure there is any benefit in forcing it into the `parse` infrastructure in the new implementation when we can just let it keep its current behavior (i.e., special-case parameters of type `itactic`).